src/shared/dissect-image.c

```diff
@@ -2562,15 +2562,10 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
         assert(m);
 
         /* As per the os-release spec, if the image is an extension it will have a file
-         * named after the image name in extension-release.d/ */
-        if (m->image_name) {
-                char *ext;
-
-                ext = strjoina("/usr/lib/extension-release.d/extension-release.", m->image_name, "0");
-                ext[strlen(ext) - 1] = '\0'; /* Extra \0 for NULSTR_FOREACH using placeholder from above */
-                paths[META_EXTENSION_RELEASE] = ext;
-        } else
-                log_debug("No image name available, will skip extension-release metadata");
+         * named after the image name in extension-release.d/ - we use the image name
+         * and try to resolve it with the extension-release helpers, as sometimes
+         * the image names are mangled on deployment and do not match anymore. */
+        paths[META_EXTENSION_RELEASE] = m->image_name;
```

Comment:
let's not shove this into paths[]. Just define a separate variable for this. This is different in *every* way after all, it's not a NULSTR, and it is used differently.